### ISSUE TITLE
Improve Windows support

### ISSUE THREAD

**Issue by adinapoli-iohk** [kazu-yamamoto/dns#82]
This is currently blocked by #80 for which I have fired a PR [here](https://github.com/kazu-yamamoto/dns/pull/81).

After #80 will be (hopefully) merged into `master`, we have now the possibility of supporting multiple nameservers. On *nix, we are exploiting this by reading all the `nameserver` entries of `/etc/resolv.conf`, but on Windows, we are not.

As you can see [here](https://github.com/kazu-yamamoto/dns/blob/master/cbits/defaultdns.c#L38-L41), we are calling [GetNetworkParams](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365968(v=vs.85).aspx) which has the following structure:

``` cpp
DWORD GetNetworkParams(
  _Out_ PFIXED_INFO pFixedInfo,
  _In_  PULONG      pOutBufLen
);
```

This returns a pointer to a [FIXED_INFO](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365900(v=vs.85).aspx) structure, defined like so:

```cpp
typedef struct {
  char            HostName[MAX_HOSTNAME_LEN + 4];
  char            DomainName[MAX_DOMAIN_NAME_LEN + 4];
  PIP_ADDR_STRING CurrentDnsServer;
  IP_ADDR_STRING  DnsServerList;
  UINT            NodeType;
  char            ScopeId[MAX_SCOPE_ID_LEN + 4];
  UINT            EnableRouting;
  UINT            EnableProxy;
  UINT            EnableDns;
} FIXED_INFO, *PFIXED_INFO;
```

In particular, `DnsServerList` is "[...]A linked list of IP_ADDR_STRING structures that specify the set of DNS servers used by the local computer.[..]".

Unfortunately, in the current implementation, we are getting only the first one and throwing away the rest. Furthermore, the current `defaultdns.c` is doing error handling with [printf](https://github.com/kazu-yamamoto/dns/blob/master/cbits/defaultdns.c#L33), which is less than ideal. In a nutshell, this ticket is about:

* Polishing the C code;
* Read the entire linked list into the Haskell world.

**Comment by kazu-yamamoto** [comment]
@adinapoli-iohk Again, please close this issue if possible.

**Comment by adinapoli-iohk** [comment]
Seems to be all working on this end (at least on my gruesome Win VM 😅 ). Thanks again @kazu-yamamoto !
